saleor/graphql/common/dataloaders.py

```diff
@@ -0,0 +1,50 @@
+from django.conf import settings
+
+from ...warehouse.models import Warehouse
+from ..account.dataloaders import AddressByIdLoader
+from ..channel.dataloaders import ChannelBySlugLoader
+from ..core.dataloaders import DataLoader
+
+
+class WarehouseCountryCodeByChannelLoader(DataLoader):
```

Comment:
Why did we extract this Dataloader to a separate module?  It is directly related to Warehouse. 
cc @fowczarek

Reply:
Because of the circular dependency. I have used `WarehouseIdLoader` in Checkout. But now i cannot do that because Warehouse has dependency to -> Channel and Channel to Checkout.

Reply:
Wouldn't import inside a function/method be better idea?

Reply:
This is considered as a bad practice AFAIK. Of course it will hide problem of huge co-dependency without visible hierarchy, but I will not do that, unless I have to.  This is exactly the same pattern as with `base_calculations.py` but there are bare functions that: acting pure functions (they are not), acting they are not holding dependency (they are). But they are in one place (what is the benefit, IDK). Later on, we have circular dependency and the only solution is to drag parameters across function calls (`if / else ` `isinstance` pattern) - to the next code explosion.

Reply:
Code is easier to maintain in the long run, when related things are in one place - that's the benefit.

Reply:
IMO we should use import inside function in this case because we have such import in a few other dataloaders. If the problem will grow, we should apply a different solution for all places where the problem occurs.

Reply:
Done, I have reverted my changes and I have used import inside `batch_load`